File: packages/emitter/testcases/common/types/generics/Generics.ts
export function identity<T>(value: T): T {
  return value;
}

// Idiomatic C# pattern: TryGet with out parameter
export function tryFirstElement<T>(arr: T[], result: out<T>): boolean {
  if (arr.length === 0) {
    return false;
  }
  result = arr[0];
  return true;
}

export class Box<T> {
  constructor(public value: T) {}

  getValue(): T {
    return this.value;
  }
}

File: packages/emitter/testcases/common/dotnet/types/generics/Generics.cs
namespace TestCases.common.types.generics
{
    public class Box<T>
    {
        public T value;

        public Box(T value)
            {
            this.value = value;
            }

        public T getValue()
            {
            return this.value;
            }
    }

            public static class Generics
            {
                public static T identity<T>(T value)
                    {
                    return value;
                    }

                public static bool tryFirstElement<T>(global::System.Collections.Generic.List<T> arr, out T result)
                    {
                    result = default;

                    if (global::Tsonic.JSRuntime.Array.length(arr) == 0)
                        {
                        return false;
                        }
                    result = global::Tsonic.JSRuntime.Array.get(arr, 0);
                    return true;
                    }
            }
}